### PR TITLE
mgr/cephadm: remove duplicate labels when adding a host

### PR DESCRIPTION
**Pull request by adk3798**
Fixes: https://tracker.ceph.com/issues/49626

Signed-off-by: Adam King <adking@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]
works! but somewhat wonder if the `HostSpec` class should be doing the handling

**Review by jmolmo — APPROVED** [review]
I like a lot the way to avoid the duplicate label!!!

File: src/pybind/mgr/cephadm/inventory.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def assert_host(self, host: str) -> None:
             raise OrchestratorError('host %s does not exist' % host)
 
     def add_host(self, spec: HostSpec) -> None:
+        if spec.labels:
+            spec.labels = list(set((spec.labels)))
         self._inventory[spec.hostname] = spec.to_json()
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/inventory.py:56]: How do we deal with clusters that are already affected by this? Maybe we should change `HostSpec.{from,to}_json()´ instead?
.

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

**Comment by adk3798** [comment]
changed this to remove duplicates in HostSpec from/to json functions rather than in the add host command.

**Review by sebastian-philipp — APPROVED** [review]
much better.

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-03-11_16:09:20-rados:cephadm-wip-swagner-testing-2021-03-11-1320-distro-basic-smithi/

Lots of errors:

* https://tracker.ceph.com/issues/49633
* https://tracker.ceph.com/issues/49740
* upgrade suite is totally broken
* `tasks.cephadm_cases.test_cli.TestCephadmCLI.test_cephfs_mirror` broke yesterday or the day before.

**Pull request merged**